### PR TITLE
feat: add a few scripts for the deployment step

### PR DESCRIPTION
**Pull request by bap2pecs**
there are some setup that we usually do manually. write scripts to automate those. this will be used in our Ansible tasks later

- add a script for creating wallets
- add a script for approve the MATIC token
- add a script for set up the committee
- support Polygon Mumbai as the L1
- fix issue where some RPCs only support replay-protected txs (EIP-155)

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -1,3 +1,12 @@
 MNEMONIC="test test test test test test test test test test test junk"
 INFURA_PROJECT_ID=""
-ETHERSCAN_API_KEY=""
\ No newline at end of file
+ETHERSCAN_API_KEY=""
+
+POLYGONSCAN_API_KEY=""
+JSONRPC_HTTP_URL=""
+DAC_URL="http://cdk-validium-data-availability:8444"
```
Comment (morganhowell95) [inline .env.example:7]: We could just handle this one via a remote config/.env like the others as well?

File: deployment/helpers/approve.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  );
+  maticTokenContractWallet = maticTokenContract.connect(signer);
+  await maticTokenContractWallet.approve(
+  // From deployments/.../deploy_output.json cdkValidiumAddress
+    process.env.CDK_VALIDIUM_ADDRESS,
+    ethers.utils.parseEther("100.0")
+  );
+}
+
+main();
```
Comment (morganhowell95) [inline deployment/helpers/approve.js]: we don't want to execute something by accident on import, so let's pass an arg or do an env check if we have the time

File: deployment/helpers/deployment-helpers.js [hunk 3]
```diff
@@ -16,12 +16,13 @@ async function deployCDKValidiumDeployer(deployerAddress, signer) {
...
     const to = '0x'; // bc deployment transaction, "to" is "0x"
     const tx = {
-        to,
-        nonce: 0,
-        value: 0,
-        gasLimit: gasLimit.toHexString(),
-        gasPrice: gasPrice.toHexString(),
-        data: deployTxCDKValidiumDeployer,
+      to,
+      chainId: network.config.chainId ?? 0, // some RPCs only support replay-protected txs (EIP-155)
```
Comment (morganhowell95) [inline deployment/helpers/deployment-helpers.js:20]: Is this comment relevant to us null coalescing to zero?  If not, let's move it somewhere else and consider throwing an exception if chain id is null

File: deployment/helpers/setup-committee.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+require("dotenv").config();
+
+const deployOutput = require("./deploymentOutput/deploy_output.json");
+const dataCommitteeContractJson = require("./compiled-contracts/CDKDataCommittee.json");
+
+async function main() {
+  const dataCommitteeContractAddress = deployOutput["cdkDataCommitteeContract"];
+  if (
+    dataCommitteeContractAddress === undefined ||
+    dataCommitteeContractAddress === ""
```
Comment (morganhowell95) [inline deployment/helpers/setup-committee.js]: !dataCommitteeContractAddress

File: deployment/helpers/wallets.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+const ethers = require("ethers");
+const fs = require('fs');
+
+async function main() {
+  const password = process.argv[2];
```
Comment (morganhowell95) [inline deployment/helpers/wallets.js:5]: const [firstArg, password] = process.argv

File: package.json [hunk 6]
```diff
@@ -91,6 +91,18 @@
     "update:genesis": "node deployment/1_createGenesis.js && node deployment/1_createGenesis.js --test --input ../docker/scripts/deploy_parameters_docker.json --out ../docker/scripts/genesis_docker.json",
     "coverage": "npx hardhat coverage",
     "gas:report": "REPORT_GAS=true npx hardhat test",
-    "gas:report:file": "rm -f .openzeppelin/unknown-31337.json && REPORT_GAS=true REPORT_GAS_FILE=true npx hardhat test"
+    "gas:report:file": "rm -f .openzeppelin/unknown-31337.json && REPORT_GAS=true REPORT_GAS_FILE=true npx hardhat test",
+
+    "saveDeployment:mumbai": "mkdir -p deployments/mumbai_$(date +%s) && cp -r deployment/deploy_*.json deployments/mumbai_$(date +%s) && cp .openzeppelin/polygon-mumbai.json deployments/mumbai_$(date +%s) && cp deployment/genesis.json deployments/mumbai_$(date +%s) && mkdir -p deploymentOutput && cp deployment/deploy_*.json deploymentOutput",
```
Comment (morganhowell95) [inline package.json:96]: let's move this to a bash script and make an `exec:script` command if we have the time, just so we can handle all save/deploy commands for all networks in one place as needed there and not pollute npm command list overtime with more complicated inline commands (ok if we just add a TODO comment for now)

**Review by morganhowell95 — APPROVED** [review]
nice work!

File: deployment/helpers/deployment-helpers.js (hunk 3 above)
Comment (morganhowell95) [reply deployment/helpers/deployment-helpers.js:20]: `network.config.chainId!`

Comment (bap2pecs) [reply deployment/helpers/deployment-helpers.js:20]: yes `network.config.chainId` can be null and it's equivalent to setting it to 0

so we cannot use `network.config.chainId!` here

File: deployment/helpers/wallets.js (hunk 5 above)
Comment (bap2pecs) [reply deployment/helpers/wallets.js:5]: the first arg is the script name iiuc which we don't need it

File: package.json (hunk 6 above)
Comment (bap2pecs) [reply package.json:96]: https://linear.app/snapchain/issue/SNA-370/extract-long-inline-npm-script-to-sh
